### ISSUE TITLE
DataGrip compatibility regression from v1.5.0 to v1.5.1

### ISSUE THREAD

**Issue by TennyZhuang** [risingwavelabs/risingwave#14102]
### Describe the bug

Although we don't have official support for DataGrip, it is generally usable in v1.5.0. However, in v1.5.1, a new bug was introduced and we should fix it.

### Error message/log

<img width="813" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/9161438/edc940bb-825d-46eb-b075-a70710648d33">

### The version of RisingWave

v1.5.1

### Additional context

#10347

**Comment by TennyZhuang** [comment]
bisecting

0bc2da252ca1402ca4c5a25fc34468f3f61bc184 passed
6da9617d43dd614a1ed5e357a2152167404a10ab failed
a5ba48bbe75a92019a1bd0d569e3fa0e5f9f5461 failed

**Comment by TennyZhuang** [comment]
#13999 
#14015

Caused the regression

**Comment by TennyZhuang** [comment]
https://github.com/pgjdbc/pgjdbc/blob/259cbe54f852818165b7cc7dfab5c8a612eff102/pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java#L2334

**Comment by TennyZhuang** [comment]
query log

```
2023-12-20T23:33:32.378972+08:00 risingwave-main ThreadId(11) mode=(extended query execute) session=1 status=ok time=0ms sql=SHOW TRANSACTION ISOLATION LEVEL
```

**Comment by TennyZhuang** [comment]
```python
import psycopg
connection = psycopg.connect(
    host="127.0.0.1",
    port=4566,
    user="root",
    password="",
    dbname="dev",
)
cursor = connection.cursor()
cursor.execute('show transaction isolation level', prepare=True)
```

```
DatabaseError: server sent data ("D" message) without prior row description ("T" message)
```

**Comment by TennyZhuang** [comment]
https://github.com/postgres/postgres/blob/dc212340058b4e7ecfc5a7a81ec50e7a207bf288/src/interfaces/libpq/fe-protocol3.c#L401

**Comment by TennyZhuang** [comment]
psql reproduce

```
dev=> show transaction isolation level; \bind \g
server sent data ("D" message) without prior row description ("T" message)
server sent data ("D" message) without prior row description ("T" message)
```

**Comment by TennyZhuang** [comment]
Conclusion: The `BeMessage::RowDescription` is generated at another place, and it's inconsistent with `rows/row_desc` in handler.

**Comment by sheltonsuen** [comment]
by the way, this may not only cause IDE error, spring application may also rise error when creating connection pool,
`Factory method 'entityManagerFactory' threw exception with message: Failed to initialize pool: Received resultset tuples, but no field structure for them`

revert to 14.0 will solve this problem

**Comment by BugenZhao** [comment]
> revert to 14.0 will solve this problem

v1.5.0 should not have such issue. Would you please give it a try?

**Comment by TennyZhuang** [comment]
https://github.com/risingwavelabs/risingwave/pull/14111

The fix will be included in v1.5.2 and v1.6.0.
